polyfill.js

```diff
@@ -4,24 +4,16 @@ if (!('fromEntries' in Object)) {
 
     const obj = {};
 
-    for (const [ index, pair ] of pairs.entries()) {
+    for (const pair of iter) {
       if (Object(pair) !== pair) {
-        // Consistent messaging to Chrome when initializing a Map:
-        throw new TypeError(`Iterator value ${ index } is not an entry object`);
+        throw new TypeError(`iterable for fromEntries should yield objects`);
       }
 
-      // Consistency with Map: object need not be technically iterable:
+      // Consistency with Map: contract is that entry has "0" and "1" keys, not
+      // that it is an array or iterable.
 
       const { 0: key, 1: val } = pair;
```

Comment:
Shouldn’t this throw if the array length isn’t 2, and if pair isn’t an array?

Reply:
I followed the contract used by the Map constructor. Although all intrinsic APIs that yield entries do so as 2-member arrays, Map is the only existing consumer of them, and all it requires is that each member is an object — not even that the keys are actually present.

```js
new Map([ {0: 'foo', 1: 'bar'}, { 1: 'baz' } ]);
// Map(2) {"foo" => "bar", undefined => "baz"}
```

Check out [step 9 in 23.1.1.1 (Map Constructor)](https://tc39.github.io/ecma262/#sec-map-iterable)

Reply:
lol, agree to be consistent.

We might want to put the “extract a list of key/value from the entry” logic in an abstract operation so that both this and Map can share it.